rtt_ros2_topics/include/orocos/rtt_ros2_topics/protocol_id.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2020 Intermodalics BVBA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OROCOS__RTT_ROS2_TOPICS__PROTOCOL_ID_H_
+#define OROCOS__RTT_ROS2_TOPICS__PROTOCOL_ID_H_
+
+// ROS 1 uses protocol id 3, so for ROS 2 we use 3 + 2 = 32 ;-)
+#ifndef ORO_ROS2_PROTOCOL_ID
+#define ORO_ROS2_PROTOCOL_ID 32
```

Comment:
This constant should not be used directly. C++ users should use the methods defined in [topics.hpp](https://github.com/orocos/rtt_ros2_integration/pull/10/files/cc53cf869f2f166c3ad9e477c3b22532176e22e5#diff-bfbdb46a0ec1da0c1f7ebd3ee74ef454) instead. And for scripting these methods are available as operations in the global service `ros` since https://github.com/orocos/rtt_ros2_integration/pull/10/commits/457b2e1d4ae908252df94c0b9aaf9beca6b5f98e.

It used to be like that already in [rtt_ros_integration](https://github.com/orocos/rtt_ros_integration), cf. [rtt_roscomm/rostopic.h:33](https://github.com/orocos/rtt_ros_integration/blob/c65cf80ce5ae07536fb5542b314f081e7e930d2d/rtt_roscomm/include/rtt_roscomm/rostopic.h#L33).

The protocol ID must be unique within an Orocos deployment. There is no world-wide registry. 0, 1 and 2 are used within RTT itself (for the default transport, CORBA and mqueue), and 3 was the `ORO_ROS_PROTOCOL_ID` defined in rtt_ros_integration for a long time now. I did not want to continue with 4 because it might be in use by other custom transports in proprietary code.